src/compas_ags/ags/core.py

```diff
@@ -105,16 +105,16 @@ def update_primal_from_dual(xy, _xy, free, fixed_x, fixed_y, i_nbrs, ij_e, _C, t
         Edge index for every vertex pair.
     _C : sparse matrix in csr format
         The connectivity matrix of the force diagram.
-    targ_l : list (optional)
+    target_lengths : list, optional
         Target lengths / target forces of the edges.
         Default is an empty list, which considers that no target lengths are considered.
-    targ_v : list (optional)
+    target_vectors : list, optional
         Target vectors of the edges.
         Default is an empty list, which considers that no target vectors are considered.
-    leaves : list
+    leaves : list, optional
         The leaves of the primal diagram.
         Default is an empty list, which considers that no leaves are considered.
-    kmax : int, optional
+    kmax : int, optional, optional
```

Comment:
unless the optionality is optional, remove the second optional

Reply:
no, the optionality is mandatory here